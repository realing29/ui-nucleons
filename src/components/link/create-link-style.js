import classNames from 'classnames/bind';
import styles from './link.scss';

const cx = classNames.bind(styles);
export const LINK_COLORS = ['black', 'dark-blue', 'blue', 'red', 'gray', 'white'];
export const EXTERNAL_COLORS = ['blue', 'white', 'black'];
export const LINK_UNDERLINE_TYPES = ['solid', 'dashed'];

/**
 * Создание стилей ссылки/псевдо-ссылки.
 * @param {string} [className=''] Пользовательские классы.
 * @param {boolean} [disableHoverEffect=false] Не реагировать при наведении.
 * @return {string} Строка с классами для ссылки/псевдо-ссылки.
 */
export const createLinkStyle = ({ className = '', disableHoverEffect = false }) =>
  cx(className, {
    link: true,
    'link-can-be-hovered': !disableHoverEffect,
  });

/**
 * Создание стилей для текста внутри ссылки/псевдо-ссылки.
 * @param {string} [color=''] Цвет текста.
 * @param {string} [underlineType=''] Подчеркивание.
 * @param {boolean} [external=false] Является ли ссылкой на внешний ресурс.
 * @return {string} Строка с классами для текста внутри ссылки/псевдо-ссылки.
 */
export const createLinkTextStyle = ({ color = '', underlineType = '', external = false }) => {
  color = LINK_COLORS.includes(color) ? color : 'blue';
  underlineType = LINK_UNDERLINE_TYPES.includes(underlineType) ? underlineType : false;
  return cx('link-text', `link-${color}`, {
    'link-external': external,
    'link-underlined': underlineType,
    [`link-underlined-${underlineType}`]: underlineType,
  });
};

/**
 * Создание стилей для иконки внешного ресурса.
 * @param {string} [color] Цвет.
 * @return {string} Строка с классами для иконки внешнего ресурса.
 */
export const createExternalStyle = color => {
  const iconColor = color && EXTERNAL_COLORS.includes(color) ? color : 'blue';
  return cx('icon-external', `external-${iconColor}`);
};
