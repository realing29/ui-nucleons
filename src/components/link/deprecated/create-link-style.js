import classNames from 'classnames/bind';
import styles from './link.scss';

const cx = classNames.bind(styles);
export const LINK_COLORS = ['black', 'dark-blue', 'blue', 'red', 'gray', 'white', 'light-gray', 'dark-gray'];
export const LINK_UNDERLINE_TYPES = ['solid', 'dashed'];

/**
 * Создание стилей ссылки/псевдо-ссылки.
 * @param {Object} params Входные параметры.
 * @param {string} [params.className=''] Пользовательские классы.
 * @param {string} [params.color=''] Цвет текста.
 * @param {string} [params.underlineType=''] Подчеркивание.
 * @param {boolean} [params.disableHoverEffect=false] Не реагировать при наведении.
 * @return {string} Строка с классами для ссылки/псевдо-ссылки.
 */
export const createLinkStyle = ({ className = '', color = '', underlineType = '', disableHoverEffect = false }) => {
  const allowedColor = LINK_COLORS.includes(color) ? color : 'blue';
  const allowedUnderlineType = LINK_UNDERLINE_TYPES.includes(underlineType) ? underlineType : false;

  return cx(className, 'link', `link-${allowedColor}`, {
    'link-can-be-hovered': !disableHoverEffect,
    'link-underlined': allowedUnderlineType,
    [`link-underlined-${allowedUnderlineType}`]: allowedUnderlineType,
  });
};
