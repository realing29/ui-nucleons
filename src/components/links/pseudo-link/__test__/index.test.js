import PseudoLink from '../';
import { shallow } from 'enzyme';
import React from 'react';
import * as styleHelpers from '../../create-link-style';

describe('<PseudoLink />', () => {
  it('calls styleHelpers with right params and renders correctly without arrow', () => {
    jest.spyOn(styleHelpers, 'createLinkStyle');
    jest.spyOn(styleHelpers, 'createLinkTextStyle');
    const pseudoLink = shallow(
      <PseudoLink
        className='testClass'
        color='black'
        underlined
        inline
        relative
        disableHoverEffect
      >
        TestPseudoLink
      </PseudoLink>
    );
    expect(styleHelpers.createLinkStyle).toHaveBeenCalledTimes(1);
    expect(styleHelpers.createLinkStyle).toHaveBeenCalledWith({
      className: 'testClass',
      disableHoverEffect: true,
      inline: true,
      relative: true,
    });
    expect(styleHelpers.createLinkTextStyle).toHaveBeenCalledTimes(1);
    expect(styleHelpers.createLinkTextStyle).toHaveBeenCalledWith({
      color: 'black',
      underlineType: 'dashed',
    });
    expect(pseudoLink).toMatchSnapshot();
  });
  it('calls styleHelpers and renders correctly with arrow', () => {
    jest.spyOn(styleHelpers, 'createLinkStyle');
    jest.spyOn(styleHelpers, 'createLinkTextStyle');
    const pseudoLink = shallow(
      <PseudoLink
        color='white'
        withArrow
        upArrow
      >
      TestPseudoLink
      </PseudoLink>
    );
    expect(styleHelpers.createLinkStyle).toHaveBeenCalledTimes(2);
    expect(styleHelpers.createLinkTextStyle).toHaveBeenCalledTimes(2);
    expect(pseudoLink).toMatchSnapshot();
  });
  it('calls functions onClick, onMouseEnter, onMouseLeave', () => {
    const onClickFunc = jest.fn(),
      onMouseEnterFunc = jest.fn(),
      onMouseLeaveFunc = jest.fn();
    const pseudoLink = shallow(
      <PseudoLink
        onClick={onClickFunc}
        onMouseEnter={onMouseEnterFunc}
        onMouseLeave={onMouseLeaveFunc}
      >
      Кнопка-блок
      </PseudoLink>
    );
    pseudoLink.simulate('click');
    expect(onClickFunc).toHaveBeenCalledTimes(1);
    pseudoLink.simulate('mouseEnter');
    expect(onMouseEnterFunc).toHaveBeenCalledTimes(1);
    pseudoLink.simulate('mouseLeave');
    expect(onMouseLeaveFunc).toHaveBeenCalledTimes(1);
  });
});
