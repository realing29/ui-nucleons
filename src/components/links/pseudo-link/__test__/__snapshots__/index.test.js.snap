// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PseudoLink /> calls styleHelpers and renders correctly with arrow 1`] = `
<span
  className="link link-can-be-hovered link-block"
>
  <span
    className="link-text link-white"
  >
    TestPseudoLink
  </span>
  <span
    className="icon-arrow"
  >
    <Icon
      color="dark-blue"
      icon={[Function]}
      inline={true}
      size={10}
    />
  </span>
</span>
`;

exports[`<PseudoLink /> calls styleHelpers with right params and renders correctly without arrow 1`] = `
<span
  className="testClass link link-relative"
>
  <span
    className="link-text link-black link-underlined link-underlined-dashed"
  >
    TestPseudoLink
  </span>
</span>
`;
